source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/DocumentValidation/Factories/ChargeCommandDocumentValidationRulesFactory.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands.Validation.DocumentValidation.ValidationRules;
+using GreenEnergyHub.Charges.Domain.Dtos.SharedDtos;
+using GreenEnergyHub.Charges.Domain.Dtos.Validation;
+using GreenEnergyHub.Charges.Domain.MarketParticipants;
+
+namespace GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands.Validation.DocumentValidation.Factories
+{
+    public class ChargeCommandDocumentValidationRulesFactory : IDocumentValidationRulesFactory<ChargeCommand>
+    {
+        private readonly IMarketParticipantRepository _marketParticipantRepository;
+
+        public ChargeCommandDocumentValidationRulesFactory(IMarketParticipantRepository marketParticipantRepository)
+        {
+            _marketParticipantRepository = marketParticipantRepository;
+        }
+
+        public async Task<IValidationRuleSet> CreateRulesAsync(ChargeCommand chargeCommand)
+        {
+            if (chargeCommand == null) throw new ArgumentNullException(nameof(chargeCommand));
```

Comment:
To avoid the SonarCloud warning is suggest to use the new `ThrowIfNull-`feature in .Net 6:
```suggestion
            ArgumentNullException.ThrowIfNull(chargeCommand);
```